drivers/gpio/gpio_shell.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <misc/printk.h>
+#include <shell/shell.h>
+#include <init.h>
+#include <string.h>
+#include <stdio.h>
+#include <gpio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <logging/log.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+
+LOG_MODULE_REGISTER(gpio_shell);
+
+struct args_index {
+	u8_t port;
+	u8_t index;
+	u8_t mode;
+	u8_t value;
+};
+
+struct args_number {
+	u8_t conf;
+	u8_t set;
+	u8_t get;
+	u8_t listen;
+};
+
+static const struct args_index args_indx = {
+	.port = 1,
+	.index = 2,
+	.mode = 3,
+	.value = 3,
+};
+
+static const struct args_number args_no = {
+	.conf = 4,
+	.set = 4,
+	.get = 3,
+	.listen = 7
+};
+
+static int cmd_gpio_conf(const struct shell *shell, size_t argc, char **argv)
+{
+	u8_t index = 0;
+	int type = GPIO_DIR_OUT;
+	struct device *dev;
+
+	if (argc == args_no.conf &&
+	    isdigit((unsigned char)argv[args_indx.index][0]) &&
+	    isalpha((unsigned char)argv[args_indx.mode][0])) {
+		index = (u8_t)atoi(argv[args_indx.index]);
+		if (!strcmp(argv[args_indx.mode], "in")) {
+			type = GPIO_DIR_IN;
+		} else if (!strcmp(argv[args_indx.mode], "out")) {
+			type = GPIO_DIR_OUT;
+		} else {
+			return 0;
+		}
+	} else {
+		shell_fprintf(shell, SHELL_NORMAL,
+			      "Wrong parameters for conf\r\n");
+		return 0;
+	}
+
+	dev = device_get_binding(argv[args_indx.port]);
+
+	if (dev != NULL) {
+		index = (uint8_t)atoi(argv[args_indx.index]);
+		LOG_INF("Configuring %s pin %d", argv[args_indx.port], index);
```

Comment:
This should printed with `shell_fprintf` because you are within a command context. Printing with LOG may mix text order on the screen. It means it will be printed after all command messages.